frontend/src/scenes/organization/Settings/inviteLogic.ts

```diff
@@ -0,0 +1,112 @@
+import { kea } from 'kea'
+import { OrganizationInviteType } from '~/types'
+import api from 'lib/api'
+import { toast } from 'react-toastify'
+import { organizationLogic } from 'scenes/organizationLogic'
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+import { inviteLogicType } from './inviteLogicType'
+import { preflightLogic } from 'scenes/PreflightCheck/logic'
+import { successToast } from 'lib/utils'
+import { router } from 'kea-router'
+import { urls } from 'scenes/urls'
+
+/** State of a single invite row (with input data) in bulk invite creation. */
+interface InviteRowState {
+    target_email: string
+    first_name: string
+    isValid: boolean
+}
+
+const EMPTY_INVITE: InviteRowState = { target_email: '', first_name: '', isValid: true }
+
+export const inviteLogic = kea<inviteLogicType<InviteRowState>>({
+    path: ['scenes', 'organization', 'Settings', 'inviteLogic'],
+    actions: {
+        updateInviteAtIndex: (payload, index: number) => ({ payload, index }),
+        deleteInviteAtIndex: (index: number) => ({ index }),
+        appendInviteRow: true,
+        resetInviteRows: true,
+    },
+    connect: {
+        values: [preflightLogic, ['preflight']],
+    },
+    reducers: {
+        invitesToSend: [
+            [EMPTY_INVITE] as InviteRowState[],
+            {
+                updateInviteAtIndex: (state, { payload, index }) => {
+                    const newState = [...state]
+                    newState[index] = { ...state[index], ...payload }
+                    return newState
+                },
+                deleteInviteAtIndex: (state, { index }) => {
+                    const newState = [...state]
+                    newState.splice(index, 1)
+                    return newState
+                },
+                appendInviteRow: (state) => [...state, EMPTY_INVITE],
+                resetInviteRows: () => [EMPTY_INVITE],
+            },
+        ],
+    },
+    selectors: {
+        canSubmit: [
+            (selectors) => [selectors.invitesToSend],
+            (invites: InviteRowState[]) =>
+                invites.filter(({ target_email }) => !!target_email).length > 0 &&
+                invites.filter(({ isValid }) => !isValid).length == 0,
+        ],
+    },
+    loaders: ({ values }) => ({
+        _invitedTeamMembers: [
```

Comment:
Why the `_`?

Reply:
I guess because this is not listed anywhere and applies just to the latest batch of invites created. Renamed to `invitedTeamMembersInternal` 🤔